### ISSUE TITLE
C# Net SDK Assign and populate metadata template. Error

### ISSUE THREAD

**Issue by mtbTahoe1** [box/box-windows-sdk-v2#385]
I am importing a large set of files from an existing document library and would like to upload the file then assign and populate an existing metadata template named (Research). The file uploads to box fine but I receive the following error when assigning the metadata template.

`Box.V2.Exceptions.BoxException: '{"message":"Instance not found for '\"template\", \"scopedKey\", \"enterprise_xxxxx\", \"Research\"'","code":"instance_tuple_not_found","request_id":"***number removed for privacy***59xxxxxx"}'`

The file upload code seems to work fine. 

`using (FileStream stream = new FileStream(@row["FilePath"].ToString(), FileMode.Open))
{
      BoxFileRequest req = new BoxFileRequest()
      {
          Name = "test12.doc",
          Parent = new BoxRequestEntity() { Id = "3***phone number removed for privacy***" }
     };
     newFile = await userClient.FilesManager.UploadAsync(req, stream);
}`

But I receive the error when I try to create the file metadata.

`var md = new Dictionary<string, object>()
                {
                    { "Symbol", "ABC" }
                };
Box.V2.Managers.BoxMetadataManager mm = userClient.MetadataManager;
var createMD = await mm.CreateFileMetadataAsync(newFile.Id.ToString(), md, "enterprise", "Research");`

**Comment by mtbTahoe1** [comment]
I  finally solved this issue. I ran the following curl call, which returned a list of all my metadata templates and their respective template keys. The template key may be different then the actual name you assign your template. In my case it was simple capitalization issue. I named my template "Research" but the key was "research" once I changed this in my code everything worked well. Hope this helps. 

curl https://api.box.com/2.0/metadata_templates/enterprise \ -H "Authorization: Bearer "YOUR DEVELOPER TOKEN"
